ls1intum/Artemis
# `Assessment`: Show average rating for exercises in course assessment dashboard

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

[ ]* - not needed

### Motivation and Context


It would beneficial for the instructors as well tutors to see the average rating of the exercise in the course.

### Description

New column in the exercise table is introduced as well as the necessary data is computed on the server and passed to the client in `for-assessment-dashboard` call.
The values in the column are `n.a.` in cases when no ratings are provided. The same logic is adjusted in the tutor-leaderboard for average score as well as average rating columns.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with Complaints enabled

1. Log in to Artemis (either as instructor or tutor)
2. Navigate to Course Administration
3. Navigate to Assessment Dashboard
4. You should be able to see a new column in the exercise table named **Average Rating**

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





no change

### Screenshots


#### Before
<img width="1263" alt="Screenshot 2021-11-08 at 8 37 57 PM" src="https://user-images.githubusercontent.com/51077603/140806641-e6874c21-42af-4869-95b8-ee77aeb2ae91.png">

-------------------------------------------

#### After
<img width="1263" src="https://user-images.githubusercontent.com/51077603/142702532-5c5611c6-0d7b-4846-917a-d4e28f1056bd.png">
